src/main/java/io/fabric8/elasticsearch/plugin/acl/DynamicACLFilter.java

```diff
@@ -345,4 +418,38 @@ public int order() {
 		return Integer.MIN_VALUE;
 	}
 	
+	private boolean uploadFile(Client tc, String filepath, String type) {
+        try (Reader reader = new FileReader(filepath)) {
+
+            final String id = tc
+                    .index(new IndexRequest("searchguard").type(type).id("0").refresh(true)
+                            .consistencyLevel(WriteConsistencyLevel.DEFAULT).source(readXContent(reader, XContentType.YAML)))
+                            .actionGet().getId();
+
+            if ("0".equals(id)) {
+                return true;
+            } else {
+                logger.warn("Configuration for '{}' failed for unknown reasons", type);
+            }
+        } catch (IOException e) {
+            logger.error("Configuration for '{}' failed due to '{}'", type, e);
+        }
+        
+        return false;
+    }
+
+    private BytesReference readXContent(final Reader reader, final XContentType xContentType) throws IOException {
```

Comment:
Apache commons has IOUtils that may be able to replace this method; its worth investigating